[Cocoa] Control whether PDF.js is included via a build-time setting

emw-apple opened the pull request:
#### d4387ce981a93c7bad97ff26700e5428c952e64a
<pre>
[Cocoa] Control whether PDF.js is included via a build-time setting
<a href="https://bugs.webkit.org/show_bug.cgi?id=241117">https://bugs.webkit.org/show_bug.cgi?id=241117</a>

Reviewed by Brent Fulgham.

Revert &quot;REGRESSION(r289850): Bundle size has increased&quot;
This reverts commit 3d8a962d9a56367e826fa968b4f1b4c8851b8982.

Expose a build-webkit feature flag `--[no-]pdfjs` which controls an
ENABLE_PDFJS build setting. This aligns with the GTK and WPE build
settings added in <a href="https://bugs.webkit.org/show_bug.cgi?id=238023.">https://bugs.webkit.org/show_bug.cgi?id=238023.</a>

On Xcode-based platforms, setting ENABLE_PDFJS causes pdfjs* resources
to be copied into WebCore, and ENABLE_PDFJS=1 to be defined.

* Source/WTF/wtf/PlatformEnable.h: For Cocoa platforms, disable
  ENABLE_PDFJS by default. It is conditionally enabled in xcconfigs.
* Source/WebCore/Configurations/WebCore.xcconfig: Recognize ENABLE_PDFJS
  setting. Define ENABLE_PDFJS as a preprocessor macro.
* Source/WebCore/Modules/pdfjs-extras/adwaita/style.css: Renamed from Source/WebCore/Modules/pdfjs/extras/adwaita/style.css.
* Source/WebCore/Modules/pdfjs-extras/cocoa/arrow.down.circle@2x.png: Renamed from Source/WebCore/Modules/pdfjs/extras/cocoa/arrow.down.circle@2x.png.
* Source/WebCore/Modules/pdfjs-extras/cocoa/minus.magnifyingglass@2x.png: Renamed from Source/WebCore/Modules/pdfjs/extras/cocoa/minus.magnifyingglass@2x.png.
* Source/WebCore/Modules/pdfjs-extras/cocoa/plus.magnifyingglass@2x.png: Renamed from Source/WebCore/Modules/pdfjs/extras/cocoa/plus.magnifyingglass@2x.png.
* Source/WebCore/Modules/pdfjs-extras/cocoa/style.css: Renamed from Source/WebCore/Modules/pdfjs/extras/cocoa/style.css.
* Source/WebCore/Modules/pdfjs-extras/content-script.js: Renamed from Source/WebCore/Modules/pdfjs/extras/content-script.js.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj: Add copy file phases
  for PDF.js source and extras. Rather than copying the entire `pdfjs`
  directory, copy members individually to avoid copying README.webkit or
  PdfJSFiles.cmake into WebCore&apos;s bundle.
* Source/WebKit/Configurations/BaseTarget.xcconfig: Define ENABLE_PDFJS
  as a preprocessor macro.
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig: Define
  ENABLE_PDFJS as a preprocessor macro.
* Tools/Scripts/webkitperl/FeatureList.pm: Add pdfjs feature.

Canonical link: <a href="https://commits.webkit.org/252343@main">https://commits.webkit.org/252343@main</a>
</pre>

emw-apple commented on Source/WTF/wtf/PlatformEnable.h:
```diff
@@ -890,7 +890,8 @@
+// FIXME: Reenable PDFJS by default for Cocoa (https://bugs.webkit.org/show_bug.cgi?id=235981).
+#if !defined(ENABLE_PDFJS) && ( /* PLATFORM(COCOA) || */ PLATFORM(GTK) || PLATFORM(WPE))
 #define ENABLE_PDFJS 1
 #endif
```
One downside to moving this preprocessor macro logic to xcconfigs is that we don't have a single "Features" xcconfig that is used by every target. So, when building with `--pdfjs`, `ENABLE(PDFJS)` will only be true in WebCore, WebKit, and WebKitLegacy, because those projects are configured to recognize the build setting. It'll be false everywhere else.

nt1m commented on Source/WTF/wtf/PlatformEnable.h:
```diff
@@ -890,7 +890,8 @@
 #endif
 
-#if !defined(ENABLE_PDFJS) && (PLATFORM(COCOA) || PLATFORM(GTK) || PLATFORM(WPE))
+// FIXME: Reenable PDFJS by default for Cocoa (https://bugs.webkit.org/show_bug.cgi?id=235981).
```
I just filed https://bugs.webkit.org/show_bug.cgi?id=242263 for this, can we reference this instead?

nt1m reviewed (commented):
Thanks for looking into this. I don't know much about the build system, so I'll let others review.

I'm noticing individual files are being specified, is it possible to specify the whole PDF.js third party folder instead to ease-up PDF.js updates? Not a deal breaker, but would be nice to have.

brentfulgham reviewed (approved):
Much cleaner!

brentfulgham commented on Source/WTF/wtf/PlatformEnable.h:
```diff
@@ -890,7 +890,8 @@
 #endif
 
-#if !defined(ENABLE_PDFJS) && (PLATFORM(COCOA) || PLATFORM(GTK) || PLATFORM(WPE))
+// FIXME: Reenable PDFJS by default for Cocoa (https://bugs.webkit.org/show_bug.cgi?id=242263).
```
Nit: Could you file a radar (or Bugzilla) for this task so we can be sure to re-enable it here?

nt1m replied:
Bugzilla: https://bugs.webkit.org/show_bug.cgi?id=242263
Radar: rdar://problem/96693948

emw-apple committed:
[Cocoa] Control whether PDF.js is included via a build-time setting
https://bugs.webkit.org/show_bug.cgi?id=241117

Reviewed by Brent Fulgham.

Revert "REGRESSION(r289850): Bundle size has increased"
This reverts commit 3d8a962d9a56367e826fa968b4f1b4c8851b8982.

Expose a build-webkit feature flag `--[no-]pdfjs` which controls an
ENABLE_PDFJS build setting. This aligns with the GTK and WPE build
settings added in https://bugs.webkit.org/show_bug.cgi?id=238023.

On Xcode-based platforms, setting ENABLE_PDFJS causes pdfjs* resources
to be copied into WebCore, and ENABLE_PDFJS=1 to be defined.

* Source/WTF/wtf/PlatformEnable.h: For Cocoa platforms, disable
  ENABLE_PDFJS by default. It is conditionally enabled in xcconfigs.
* Source/WebCore/Configurations/WebCore.xcconfig: Recognize ENABLE_PDFJS
  setting. Define ENABLE_PDFJS as a preprocessor macro.
* Source/WebCore/Modules/pdfjs-extras/adwaita/style.css: Renamed from Source/WebCore/Modules/pdfjs/extras/adwaita/style.css.
* Source/WebCore/Modules/pdfjs-extras/cocoa/arrow.down.circle@2x.png: Renamed from Source/WebCore/Modules/pdfjs/extras/cocoa/arrow.down.circle@2x.png.
* Source/WebCore/Modules/pdfjs-extras/cocoa/minus.magnifyingglass@2x.png: Renamed from Source/WebCore/Modules/pdfjs/extras/cocoa/minus.magnifyingglass@2x.png.
* Source/WebCore/Modules/pdfjs-extras/cocoa/plus.magnifyingglass@2x.png: Renamed from Source/WebCore/Modules/pdfjs/extras/cocoa/plus.magnifyingglass@2x.png.
* Source/WebCore/Modules/pdfjs-extras/cocoa/style.css: Renamed from Source/WebCore/Modules/pdfjs/extras/cocoa/style.css.
* Source/WebCore/Modules/pdfjs-extras/content-script.js: Renamed from Source/WebCore/Modules/pdfjs/extras/content-script.js.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj: Add copy file phases
  for PDF.js source and extras. Rather than copying the entire `pdfjs`
  directory, copy members individually to avoid copying README.webkit or
  PdfJSFiles.cmake into WebCore's bundle.
* Source/WebKit/Configurations/BaseTarget.xcconfig: Define ENABLE_PDFJS
  as a preprocessor macro.
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig: Define
  ENABLE_PDFJS as a preprocessor macro.
* Tools/Scripts/webkitperl/FeatureList.pm: Add pdfjs feature.

Canonical link: https://commits.webkit.org/252343@main
